File: examples/Example_11.java
/*
 * Example_11.java
 *
 * Copyright (c) 2026 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package examples;

import java.io.*;
import com.pdfjet.*;
import com.pdfjet.fonts.*;
import com.pdfjet.barcodes.*;

/**
 * Example_11.java
 * This example draws a Code 128, a Code 39, a UPC-A and an EAN-13 barcode,
 * each next to a label, and then barcodes drawn from top to bottom and from
 * bottom to top.
 */
public class Example_11 {
    public Example_11() throws Exception {
        PDF pdf = new PDF(
            new BufferedOutputStream(new FileOutputStream("Example_11.pdf")));
        pdf.setCompliance(Compliance.PDF_UA_1);
        pdf.setTitle("Linear Barcodes");

        Font f1 = new Font(pdf, IBMPlexSans.Regular);
        f1.setSize(12f);

        Font f2 = new Font(pdf, IBMPlexSans.SemiBold);
        f2.setSize(12f);

        Page page = new Page(pdf, Letter.PORTRAIT);

        TextLine text = new TextLine(f2, "Linear Barcodes");
        text.setFontSize(22f);
        text.setLocation(70f, 80f);
        text.drawOn(page);

        String[] labels = {
            "Code 128",
            "Code 39",
            "UPC-A",
            "EAN-13",
        };
        String[] notes = {
            "Letters, digits and symbols",
            "Upper case letters and digits",
            "11 digits, the check digit is added",
            "12 digits, the check digit is added",
        };
        Barcode[] barcodes = {
            new Barcode(Barcode.CODE_128, "Hellö, World!"),
            new Barcode(Barcode.CODE_39, "WIKIPEDIA"),
            new Barcode(Barcode.UPC_A, "51234567890"),
            new Barcode(Barcode.EAN_13, "051234567890"),
        };
        // UPC-A and EAN-13 need wider bars for the digits under them.
        float[] moduleLengths = {0.75f, 0.75f, 1f, 1f};

        float y = 130f;
        for (int i = 0; i < barcodes.length; i++) {
            new TextLine(f2, labels[i]).setLocation(70f, y + 15f).drawOn(page);
            TextLine note = new TextLine(f1, notes[i]);
            note.setFontSize(10f);
            note.setTextColor(Color.gray);
            note.setLocation(70f, y + 32f);
            note.drawOn(page);

            Barcode barcode = barcodes[i];
            barcode.setLocation(290f, y);
            barcode.setModuleLength(moduleLengths[i]);
            barcode.setFont(f1);
            float[] xy = barcode.drawOn(page);
            y = xy[1] + 30f;
        }

        // The same barcodes can be drawn from top to bottom and from bottom to top.
        new TextLine(f2, "Vertical barcodes").setLocation(70f, y + 15f).drawOn(page);

        Barcode barcode = new Barcode(Barcode.CODE_128, "G86513JVW0C");
        barcode.setLocation(70f, y + 35f);
        barcode.setModuleLength(0.75f);
        barcode.setDirection(Direction.TOP_TO_BOTTOM);
        barcode.setFont(f1);
        float[] xy = barcode.drawOn(page);

        barcode = new Barcode(Barcode.CODE_39, "CODE39");
        barcode.setLocation(xy[0] + 60f, y + 35f);
        barcode.setModuleLength(0.75f);
        barcode.setDirection(Direction.BOTTOM_TO_TOP);
        barcode.setFont(f1);
        xy = barcode.drawOn(page);

        barcode = new Barcode(Barcode.EAN_13, "051234567890");
        barcode.setLocation(xy[0] + 60f, y + 35f);
        barcode.setModuleLength(1f);
        barcode.setDirection(Direction.BOTTOM_TO_TOP);
        barcode.setFont(f1);
        barcode.drawOn(page);

        pdf.complete();
    }

    public static void main(String[] args) throws Exception {
        long time0 = System.currentTimeMillis();
        new Example_11();
        long time1 = System.currentTimeMillis();
        System.out.printf("Example_11 => %4d ms%n", time1 - time0);
    }
}   // End of Example_11.java

File: examples/Example_11.cs
using System;
using System.IO;
using System.Diagnostics;
using PDFjet.NET;

/**
 *  Example_11.cs
 */
public class Example_11 {
    public Example_11() {
        PDF pdf = new PDF( new BufferedStream(
                new FileStream("Example_11.pdf", FileMode.Create)));

        Font f1 = new Font(pdf, "fonts/OpenSans/OpenSans-Regular.ttf.stream");

        Page page = new Page(pdf, Letter.PORTRAIT);

        Barcode code = new Barcode(Barcode.CODE128, "Hellö, World!");
        code.SetLocation(170f, 70f);
        code.SetModuleLength(0.75f);
        code.SetFont(f1);
        code.DrawOn(page);

        code = new Barcode(Barcode.CODE128, "G86513JVW0C");
        code.SetLocation(170f, 170f);
        code.SetModuleLength(0.75f);
        code.SetDirection(Barcode.TOP_TO_BOTTOM);
        code.SetFont(f1);
        code.DrawOn(page);

        code = new Barcode(Barcode.CODE39, "WIKIPEDIA");
        code.SetLocation(270f, 370f);
        code.SetModuleLength(0.75f);
        code.SetFont(f1);
        code.DrawOn(page);

        code = new Barcode(Barcode.CODE39, "CODE39");
        code.SetLocation(400f, 70f);
        code.SetModuleLength(0.75f);
        code.SetDirection(Barcode.TOP_TO_BOTTOM);
        code.SetFont(f1);
        code.DrawOn(page);

        code = new Barcode(Barcode.CODE39, "CODE39");
        code.SetLocation(450f, 70f);
        code.SetModuleLength(0.75f);
        code.SetDirection(Barcode.BOTTOM_TO_TOP);
        code.SetFont(f1);
        code.DrawOn(page);

        code = new Barcode(Barcode.UPC, "712345678904");
        code.SetLocation(450f, 270f);
        code.SetModuleLength(0.75f);
        code.SetDirection(Barcode.BOTTOM_TO_TOP);
        code.SetFont(f1);
        code.DrawOn(page);

        pdf.Complete();
    }

    public static void Main(String[] args) {
        Stopwatch sw = Stopwatch.StartNew();
        long time0 = sw.ElapsedMilliseconds;
        new Example_11();
        long time1 = sw.ElapsedMilliseconds;
        sw.Stop();
        TextUtils.PrintDuration("Example_11", time0, time1);
    }
}   // End of Example_11.cs
